src/api/queries/pipelines.ts

```diff
@@ -163,44 +169,59 @@ export const useCreateTektonResourcesMutation = (
 export const useStartPipelineRunMutation = (
   pipelineGroup: CranePipelineGroup,
   action: CranePipelineAction,
+  options?: Partial<UseMutationOptions>,
 ) => {
   const [pipelineRunModel] = useK8sModel(pipelineRunGVK);
-  return useMutation([pipelineGroup.name, action], () => {
-    const pipeline = pipelineGroup.pipelines[action];
-    const latestPipelineRun = pipelineGroup.pipelineRuns[action][0];
-    if (!pipeline || !latestPipelineRun) return Promise.reject('Pipeline or PipelineRun not found');
-    if (latestPipelineRun.spec.status === 'PipelineRunPending') {
-      return k8sPatch({
+  return useMutation(
+    [pipelineGroup.name, action],
+    () => {
+      const pipeline = pipelineGroup.pipelines[action];
+      const latestPipelineRun = pipelineGroup.pipelineRuns[action][0];
+      if (!pipeline || !latestPipelineRun)
+        return Promise.reject('Pipeline or PipelineRun not found');
+      if (latestPipelineRun.spec.status === 'PipelineRunPending') {
+        return k8sPatch({
+          model: pipelineRunModel,
+          resource: latestPipelineRun,
+          data: [{ op: 'remove', path: '/spec/status' }],
+        });
+      }
+      const newPipelineRun: CranePipelineRun = {
+        apiVersion: 'tekton.dev/v1beta1',
+        kind: 'PipelineRun',
+        spec: { ...latestPipelineRun.spec },
+        metadata: {
+          generateName: `${pipeline.metadata?.name || ''}-`,
```

Comment:
That's a habit of mine for places where TS requires a defined string and I know it will always be a defined string, but technically the type can be undefined. (these pipelines will always have metadata). I maybe should be using `as string` instead for those cases... something to think about.

But in this case I don't think it's actually required, I'll remove it.